packages/gatsby-image/src/index.js

```diff
@@ -168,10 +154,16 @@ class Image extends React.Component {
     this.handleRef = this.handleRef.bind(this)
   }
 
+  componentDidMount() {
+    if (this.props.critical) {
+      this.setState({imgLoaded: true});
```

Comment:
onLoad can not be relied on to change the state for images that where loaded from the DOM

Reply:
What did you change here? If you push individual commits, that's helpful to people reviewing your code.

BTW, did you see https://github.com/gatsbyjs/gatsby/issues/4297#issuecomment-410994261

Reply:
Here I fixed up the lint error about spaces around braces.